R/dup.R

```diff
@@ -5,14 +5,29 @@
 #'
 #' @inheritParams str_detect
 #' @param times Number of times to duplicate each string.
+#' @param sep String to insert between each duplicate.
 #' @return A character vector the same length as `string`/`times`.
 #' @export
 #' @examples
 #' fruit <- c("apple", "pear", "banana")
 #' str_dup(fruit, 2)
 #' str_dup(fruit, 1:3)
 #' str_c("ba", str_dup("na", 0:5))
-str_dup <- function(string, times) {
+str_dup <- function(string, times, sep = NULL) {
   vctrs::vec_size_common(string = string, times = times)
-  stri_dup(string, times)
+  if(is.null(sep)){
+    stri_dup(string, times)
+  } else {
+  # stri_dup does not currently support sep
+  check_string(sep)
+  lapply(seq_along(string), function(i) {
```

Comment:
Could you use `map_chr()` here?

Reply:
I might be being a bit dim, but I couldn't manage to get this to work. The problem I was hitting was that map_chr wants to return a length 1 vector.